### PR TITLE
Add Rate Limiting to Login Page

### PR DESCRIPTION
**Pull request by Maffooch**
Dependent on #4783

Add the ability to rate limit the login page. 

Uses https://github.com/jsocol/django-ratelimit

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Did you look into if / how this would work if you have multiple django containers/pods or when using multiple uwsgi processes? I think we need to explain in the docs what works and doesn't work.

**Comment by Maffooch** [comment]
Didn't know that was even possible. How can I go about setting up an environment to test that?

**Comment by valentijnscholten** [comment]
The docker compose in release mode already has 2 processes: https://github.com/valentijnscholten/django-DefectDojo/blob/d30234cba2278b73848c8f1f85362a0adbd24c21/docker/entrypoint-uwsgi.sh#L36
I _think_ the default cache used by the ratelimiter is not cross-process, typically people use some external cache like memcached or redis for that.

**Comment by Maffooch** [comment]
It appears to block requests on a per process basis (guessing based on the 500 followed by 2 200's)
image

Here is what the documentation has to say about caching
```
After activating django-ratelimit, you should ensure that your cache backend is setup to be both persistent and work across multiple deployment worker instances (for instance UWSGI workers). Read more in the Django docs on caching.
```

Does dojo use any sort of caching method out of the box?

**Comment by valentijnscholten** [comment]
No, it only uses the `default` cache, which is in memory and local to a process/worker. That's also the reason why we can't easily add query caching for example or other forms of global caching. If we were to switch to `redis` as default celery broker, we could also use `redis` as a cross worker/process cache. I think for it's best to just make the limitation very clear in our docs.
Maybe something could be achieved by doing sticky sessions in `nginx`.

File: docs/content/en/integrations/rate_limiting.md [hunk 1]
```diff
@@ -46,6 +46,10 @@ In the event of a brute force attack, a users credentials could potentially be c
 
 In an attempt to circumvent that event, setting `DD_RATE_LIMITER_ACCOUNT_LOCKOUT` will force a user to reset their password upon the next attempted login. 
 
+#### Multi-Process Behvaior
```
Comment (valentijnscholten) [inline docs/content/en/integrations/rate_limiting.md:49]:
```suggestion
#### Multi-Process Behavior
```

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

**Review by devGregA — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved.

**Pull request merged**